core/trino-main/src/main/java/io/trino/metadata/Metadata.java

```diff
@@ -669,4 +669,19 @@ default ResolvedFunction getCoercion(Type fromType, Type toType)
      * Get the target table handle after performing redirection.
      */
     RedirectionAwareTableHandle getRedirectionAwareTableHandle(Session session, QualifiedObjectName tableName);
+
+    /**
+     * Get the target table handle after performing redirection with a table version.
+     */
+    RedirectionAwareTableHandle getRedirectionAwareTableHandle(Session session, QualifiedObjectName tableName, Optional<TableVersionHandle> version);
+
+    /**
+     * Returns a version handle given a specified table name, travel type, version type, and version value.
+     */
+    Optional<TableVersionHandle> getVersionHandle(Session session, QualifiedObjectName tableName, TableVersion version);
```

Comment:
I'm not sure we need the ConnectorVersionHandle abstraction. It adds a level of indirection that connectors need to implement explicitly. Instead, I'm thinking we should have two methods:

* get table at a point in time
* get table at a specific version

Let me think a bit more about it.

Reply:
Kind of a continuation of the other two comments. Originally I had implemented temporal time travel in the engine along those lines, but when I started implementing version time travel, I migrated both over to the VersionHandle abstraction. I want to keep the door open for future engine optimizations and enhancements (such as range support) that use the version token returned via this SPI. I'd love to hear your thoughts though.